lib/spack/docs/workflows.rst

```diff
@@ -1028,6 +1028,85 @@ or filesystem views.  However, it has some drawbacks:
    integrate Spack explicitly in their workflow.  Not all users are
    willing to do this.
 
+------------------------
+Using Spack on Travis-CI
+------------------------
+
+Spack can be deployed as provider for userland software in
+`Travis-CI <https://http://travis-ci.org>`_.
+
+A starting-point for a ``.travis.yml`` file can look as follows.
+It uses `caching <https://docs.travis-ci.com/user/caching/>`_ for
+already build environments, so make sure to clean the Travis cache if
+you run into problems.
+
+The main points that are implemented below:
+
+#. Travis is detected as having up to 34 cores available, but only 2
+   are actually allocated for the user. We limit the parallelism of
+   the spack builds in the config.
+   (The Travis yaml parser is a bit buggy on the echo command.)
+
+#. Builds over 10 minutes need to be prefixed with ``travis_wait``.
+   Alternatively, generate output once with ``spack install -v``.
+
+#. Travis builds are non-interactive. This prevents using bash
+   aliases and functions for modules. We fix that by sourcing
+   ``/etc/profile`` first (or running everything in a subshell with
+   ``bash -l -c '...'``).
+
+.. code-block:: yaml
+
+   language: cpp
+   sudo: false
+   dist: trusty
+
+   cache:
+     apt: true
+     directories:
+       - $HOME/.cache
+
+   addons:
+     apt:
+       sources:
+         - ubuntu-toolchain-r-test
+       packages:
+         - g++-4.9
+         - environment-modules
+
+   env:
+     global:
+       - SPACK_ROOT: $HOME/.cache/spack
+       - PATH: $PATH:$HOME/.cache/spack/bin
+
+   before_install:
+     - export CXX=g++-4.9
+     - export CC=gcc-4.9
+     - export FC=gfortran-4.9
+     - export CXXFLAGS="-std=c++11"
+
+   install:
+     - SPACK_FOUND=$(which spack >/dev/null && { echo 0; } || { echo 1; })
+     - if [ $SPACK_FOUND -ne 0 ]; then
```

Comment:
so
```yaml
- if which spack > /dev/null ; then
    ...;
  fi
````

I modified this from a `g++ --version` check of mine, where I couldn't use a return code ;D